Index: yii/db/schema/CDbCriteria.php
===================================================================
--- yii/db/schema/CDbCriteria.php	(revisión: 236)
+++ yii/db/schema/CDbCriteria.php	(revisión: 242)
@@ -189,12 +189,12 @@
 		{
 			if($condition===array())
 				return $this;
-			$condition='('.implode(') '.$operator.' (',$condition).')';
+			$condition=''.implode(' '.$operator.' ',$condition).'';
 		}
 		if($this->condition==='')
 			$this->condition=$condition;
 		else
-			$this->condition='('.$this->condition.') '.$operator.' ('.$condition.')';
+			$this->condition=''.$this->condition.' '.$operator.' '.$condition.'';
 		return $this;
 	}
 
Index: yii/web/helpers/CHtml.php
===================================================================
--- yii/web/helpers/CHtml.php	(revisión: 251)
+++ yii/web/helpers/CHtml.php	(revisión: 261)
@@ -1275,9 +1275,10 @@
 		self::resolveNameID($model,$attribute,$htmlOptions);
 		// add a hidden field so that if a model only has a file field, we can
 		// still use isset($_POST[$modelClass]) to detect if the input is submitted
-		$hiddenOptions=isset($htmlOptions['id']) ? array('id'=>self::ID_PREFIX.$htmlOptions['id']) : array('id'=>false);
-		return self::hiddenField($htmlOptions['name'],'',$hiddenOptions)
-			. self::activeInputField('file',$model,$attribute,$htmlOptions);
+                 
+		//$hiddenOptions=isset($htmlOptions['id']) ? array('id'=>self::ID_PREFIX.$htmlOptions['id']) : array('id'=>false);
+		return //self::hiddenField($htmlOptions['name'],'',$hiddenOptions)
+			 self::activeInputField('file',$model,$attribute,$htmlOptions);
 	}
 
 	/**
Index: yii/zii/widgets/assets/gridview/jquery.yiigridview.js
===================================================================
--- yii/zii/widgets/assets/gridview/jquery.yiigridview.js	(revisión: 302)
+++ yii/zii/widgets/assets/gridview/jquery.yiigridview.js	(copia de trabajo)
@@ -78,7 +78,7 @@
 				gridSettings[id] = settings;
 
 				if (settings.ajaxUpdate.length > 0) {
-					$(document).on('click', settings.updateSelector, function () {
+					$(document).off('click', settings.updateSelector).on('click', settings.updateSelector, function () {
 						$('#' + id).yiiGridView('update', {url: $(this).attr('href')});
 						return false;
 					});
